lib/tasks/export.rake

```diff
@@ -171,4 +171,34 @@ namespace :export do
       break # Can't do much about this, so just break
     end
   end
+
+  desc "Export command papers"
+  task command_papers: :environment do
+    puts 'Fetching command papers...'
+    path = "tmp/command-papers-#{Time.now.to_i}.csv"
+    puts "Generating CSV in #{path}..."
+
+    CSV.open(path, 'w') do |csv|
+      csv << [
+        'Title',
+        'Command Paper Number',
+        'URL'
+      ]
+
+      Attachment.where("command_paper_number <> ''").all.each do |file|
+        begin
+          url = file&.attachable.try(:search_link)
+
+          csv << [
+            file.title,
+            file.command_paper_number,
+            url ? "https://gov.uk#{url}" : file.url
+          ]
+        rescue StandardError => e
+          puts e
+          next
```

Comment:
One gets ~8 errors when running this. Likely there are some command papers that have bad relations (`attachment_data` not being defined isn't good), but I think these can be ignored.

The errors:

```
undefined method `slug' for nil:NilClass
undefined method `slug' for nil:NilClass
FileAttachment#url delegated to attachment_data.url, but attachment_data is nil: #<FileAttachment id: 2190313, created_at: "2017-07-18 08:55:19", updated_at: "2017-07-18 08:55:19", title: "Senior Salaries Review Body supplementary report 2...", accessible: false, isbn: "9781474145572", unique_reference: "", command_paper_number: "Cm 9456", order_url: "", price_in_pence: nil, attachment_data_id: 629659, ordering: 2, hoc_paper_number: "", parliamentary_session: "", unnumbered_command_paper: false, unnumbered_hoc_paper: false, attachable_id: 748447, attachable_type: "Edition", type: "FileAttachment", slug: nil, locale: nil, external_url: nil, content_id: "a3376f30-7fae-4e53-9236-6464f9002a16", deleted: true, print_meta_data_contact_address: nil, web_isbn: nil>
FileAttachment#url delegated to attachment_data.url, but attachment_data is nil: #<FileAttachment id: 2190314, created_at: "2017-07-18 08:56:58", updated_at: "2017-07-18 08:56:58", title: "Senior Salaries Review Body supplementary report 2...", accessible: false, isbn: "9781474145589", unique_reference: "", command_paper_number: "Cm 9456", order_url: "", price_in_pence: nil, attachment_data_id: 629660, ordering: 3, hoc_paper_number: "", parliamentary_session: "", unnumbered_command_paper: false, unnumbered_hoc_paper: false, attachable_id: 748447, attachable_type: "Edition", type: "FileAttachment", slug: nil, locale: nil, external_url: nil, content_id: "db9f266d-2abb-4d23-b8ee-698a445c6578", deleted: true, print_meta_data_contact_address: nil, web_isbn: nil>
FileAttachment#url delegated to attachment_data.url, but attachment_data is nil: #<FileAttachment id: 2190454, created_at: "2017-07-18 09:26:54", updated_at: "2017-07-18 09:26:54", title: "Police Remuneration Review Body report: 2017 (web ...", accessible: false, isbn: "9781474146487", unique_reference: "", command_paper_number: "Cm 9453", order_url: "", price_in_pence: nil, attachment_data_id: 629695, ordering: 0, hoc_paper_number: "", parliamentary_session: "", unnumbered_command_paper: false, unnumbered_hoc_paper: false, attachable_id: 748488, attachable_type: "Edition", type: "FileAttachment", slug: nil, locale: nil, external_url: nil, content_id: "d65fa540-d4fb-4dc2-8af2-3d6e0a5e1770", deleted: true, print_meta_data_contact_address: nil, web_isbn: nil>
undefined method `slug' for nil:NilClass
undefined method `slug' for nil:NilClass
undefined method `slug' for nil:NilClass
```